app/src/main/java/com/lucas/medicaltools/MainActivity.kt

```diff
@@ -1,11 +1,87 @@
 package com.lucas.medicaltools
 
+import android.annotation.SuppressLint
+import android.app.SearchManager
+import android.content.Context
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.view.Menu
+import android.view.View
+import android.widget.ProgressBar
+import androidx.appcompat.widget.SearchView
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.lucas.medical_equip.repository.MedicalTool
+import com.lucas.medical_equip.service.RetrofitBuilder
+import com.lucas.medicaltools.databinding.ActivityMainBinding
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import io.reactivex.rxjava3.schedulers.Schedulers
+import mu.KotlinLogging
+import java.util.*
 
+private val logger = KotlinLogging.logger {}
 class MainActivity : AppCompatActivity() {
```

Comment:
Remember to implement your view (MainView.kt) in the Mosby way
This would mean you have to implement your render function along with each intent method.